Update jobs repo notebook docs (#23)
databricks/run-notebook: README.md

See [action.yml](action.yml) for the latest interface and docs.

> ### Run notebook within a temporary checkout of the current Repo

Reviewer: ```suggestion
### (Recommended) Run notebook within a temporary checkout of the current Repo
```

Mark this option as recommended? Since with the Jobs repos integration this is now our recommended approach

            -d 'grant_type=client_credentials' \
            -d 'scope=2ff814a6-3304-4ab8-85cb-cd0e6f879c1d%2F.default' \
            -d 'client_secret=${{ secrets.AZURE_SP_CLIENT_SECRET }}' |  jq -r  '.access_token')" >> $GITHUB_ENV
      - name: Trigger model training notebook from PR branch
        uses: databricks/run-notebook@v0
        with:

>           # Run our notebook against a remote repo

Reviewer: ```suggestion
```
This line (I know it was there before) seems a little confusing / not necessary (what does it mean to "run the notebook against a remote repo"?)

          local-notebook-path: notebooks/deployments/MainNotebook

>           # If this is triggered from a PR use the PR's head commit, otherwise use github.sha.

Reviewer: ```suggestion
          # If the current workflow is triggered from a PR, run notebook code from the PR's head commit, otherwise use github.sha.
```
